packages/flutter/test/widgets/tap_region_test.dart

```diff
@@ -1053,4 +1053,101 @@ void main() {
     await click(find.text('Outside Surface'));
     expect(tappedInside, isEmpty);
   });
+
+  testWidgets('TapRegion should register and unregister correctly on push and pop', (WidgetTester tester) async {
+    const ValueKey<String> tapRegion1Key = ValueKey<String>('TapRegion');
+    const ValueKey<String> tapRegion2Key = ValueKey<String>('TapRegion2');
+
+    int count1 = 0;
+    int count2 = 0;
+
+    final TapRegion tapRegion1 = TapRegion(
+      key: tapRegion1Key,
+      onTapOutside: (PointerEvent event) {
+        count1++;
+      },
+      child: Container(
+        width: 100,
+        height: 100,
+        color: Colors.red,
+      ),
+    );
+
+    final TapRegion tapRegion2 = TapRegion(
+      key: tapRegion2Key,
+      onTapOutside: (PointerEvent event) {
+        count2++;
+      },
+      child: Container(
+        width: 100,
+        height: 100,
+        color: Colors.red,
+      ),
+    );
+
+    Widget buildPage(Widget body, {FloatingActionButton? fab}) {
+      return Scaffold(
+        appBar: AppBar(
+          backgroundColor: Colors.blue,
+          title: const Text('Test Page'),
+        ),
+        body: Center(child: body),
+        floatingActionButton: fab,
+      );
+    }
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: buildPage(
+          tapRegion1,
+          fab: FloatingActionButton(
+            onPressed: () {
+              Navigator.push(
+                tester.element(find.byType(FloatingActionButton)),
+                MaterialPageRoute<Widget>(builder: (BuildContext context) => buildPage(tapRegion2)),
+              );
+            },
+            child: const Icon(Icons.add),
+          ),
+        ),
+      ),
+    );
+
+    await tester.pumpAndSettle();
+
+    // Tap outside the first TapRegion to trigger onTapOutside.
+    await tapOutside(tester, find.byKey(tapRegion1Key));
+    expect(count1, 1);
+    expect(count2, 0);
+
+    await tester.tap(find.byType(FloatingActionButton));
+    await tester.pumpAndSettle();
+
+    // Tap outside the second TapRegion to trigger onTapOutside
+    await tapOutside(tester, find.byKey(tapRegion2Key), times: 10);
+    expect(count1, 2); // When the Fab is pressed, the first TapRegion is still active.
+    expect(count2, 10);
+
+
+
+    // Back to the first page.
+    Navigator.pop(tester.element(find.byType(Scaffold).last));
+    await tester.pumpAndSettle();
+
+    // Tap outside the first TapRegion to trigger onTapOutside
+    await tapOutside(tester, find.byKey(tapRegion1Key));
+    expect(count1, 3);
+    expect(count2, 10);
+  });
+}
+
+Future<void> tapOutside(WidgetTester tester, Finder regionFinder, {int times = 1}) async {
```

Comment:
Why do we need to tap N times here? Isn't just tapping once or twice enough?

Reply:
I can remove it. It was originally added for an initial test, but I kept it without much consideration.